packages/core/src/tools/utils.ts

```diff
@@ -156,24 +156,32 @@ export function noop() {}
  *
  * Note: this still assumes that JSON.stringify is correct.
  */
-export function jsonStringify(
-  value: unknown,
-  replacer?: Array<string | number>,
-  space?: string | number
-): string | undefined {
+export function jsonStringify(value: unknown, space?: string | number): string | undefined {
   if (typeof value !== 'object' || value === null) {
     return JSON.stringify(value)
   }
 
-  // Note: The order matter here. We need to detach toJSON methods on parent classes before their
+  // Note: The order matters here. We need to detach toJSON methods on parent classes before their
   // subclasses.
   const restoreObjectPrototypeToJson = detachToJsonMethod(Object.prototype)
   const restoreArrayPrototypeToJson = detachToJsonMethod(Array.prototype)
   const restoreValuePrototypeToJson = detachToJsonMethod(Object.getPrototypeOf(value))
   const restoreValueToJson = detachToJsonMethod(value)
 
+  // Line below can be removed when migrating to TS4.8+
+  const isValidObject = (v: any): v is object => typeof v === 'object' && v !== null
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏use `getType(destination) === 'object'` instead